controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -642,7 +642,7 @@ private HttpResponse services(String tenantName, String applicationName, String
         ApplicationView applicationView = controller.getApplicationView(tenantName, applicationName, instanceName, environment, region);
         ServiceApiResponse response = new ServiceApiResponse(ZoneId.from(environment, region),
                                                              new ApplicationId.Builder().tenant(tenantName).applicationName(applicationName).instanceName(instanceName).build(),
-                                                             controller.zoneRegistry().getConfigServerUris(ZoneId.from(environment, region)),
+                                                             controller.zoneRegistry().getConfigServerApiUris(ZoneId.from(environment, region)),
```

Comment:
I think this is wrong. The API should never return direct/VIP config server URLs, as those are not usable by clients anyway due to certificate mismatch.

It should instead return the appropriate URL under `/zone/v2/` so that the final URL is becomes `controller-host:port/zone/v2/<env>/<region>/serviceview/v1/tenant/...`.

Reply:
Never mind, I see now that `ServiceApiResponse` rewrites it to another path in `/application/v4`.

However, it should be possible to rewrite the URLs  (and simplify the code) without requiring knowledge of the config server URLs.